pkg/sqlite/load.go

```diff
@@ -1427,16 +1419,20 @@ func (s *sqlLoader) DeprecateBundle(path string) error {
 	}
 
 	for _, bundle := range tailBundles {
-		err = s.rmChannelEntry(tx, bundle)
-		if err != nil {
+		if err := s.rmChannelEntry(tx, bundle); err != nil {
 			return err
 		}
-		err := s.rmBundle(tx, bundle)
-		if err != nil {
+		if err := s.rmBundle(tx, bundle); err != nil {
 			return err
 		}
```

Comment:
I don't think you're changing this behavior in this PR but...

It looks like we include `skips` in the tail bundles without verifying that they are all non-channel-heads. Isn't it possible to have:
```
package foo, default channel: alpha

channel: alpha, head: v2
channel: beta, head: v4

v4 replaces: v3
v3 replaces: v1, skips: v2
v2 replaces: v1
```

If I'm understanding correctly, when we deprecate v3, we would remove v1 (fine) and v2 (not okay because it is the head of the alpha channel, which is the default channel)

Reply:
good catch.

It's probably simplest to add a queue here and perform a breadth-first search for channel heads.